### ISSUE TITLE
Fit semi-supervised on test data?

### ISSUE THREAD

**Issue by LouisCarpentier42** [ML-KULeuven/dtaianomaly#59]
In the ``Workflow``, there is a flag to fit unsupervised detectors on the test data directly (``fit_unsupervised_on_test_data``). This is no problem as the methods are unsupervised and thus do not use any ground truth labels. 

Would it be possible to add a similar flag for semi-supervised methods (``fit_semi_supervised_on_test_data``)? For these methods, a train set consisting of normal data is used to avoid learning anomalous behavior. However, it could also be possible to train directly on the test set (ignoring that the trainings data should be normal), and then also predict on this same set. This way, you can also analyze the robustness of a semi-supervised anomaly detector against anomalies in the training data. 

This should be a relatively easy addition. The main question is whether this functionality is desirable or not.
